crm/server-scripts/crm/Lead/UpdateCampaigns.js

```diff
@@ -0,0 +1,39 @@
+export default {
+  label: 'Update Campaigns after creation or update',
+  description: 'Update related Campaigns when a Lead is created or updated',
+
+  * triggers ({ after }) {
+    yield after('create', 'update')
+      .for('compose:record')
+      .where('module', 'Lead')
+      .where('namespace', 'crm')
+  },
+
+  async exec ({ $record }, { Compose }) {
+    // Update campaign record
+    if ($record.values.CampaignId) {
+      const filter = $record.values.CampaignId.map(cID => {
+        return `recordID = '${cID}'`
+      }).join(' OR ')
+
+      if (filter) {
+        await Compose.findRecords(filter, 'Campaigns')
+          .catch(() => ({ set: [] }))
+          .then(({ set }) => {
+            Promise.all(set.map(campaign => {
+              // Find all Leads that have the same Campaign
+              const { filter } = await Compose.findRecords(`CampaignId = '${campaign.recordID}'`, 'Lead')
+                .catch(() => ({ filter: { count: 0 } }))
+
+              campaign.values.NumberOfLeads = filter.count
+              campaign.values.NumberOfConvertedLeads = set.filter(l => l.values.IsConverted).length || 0
```

Comment:
Two things:

1. The `set` here contains campaigns, not leads (am I missing something?)
2. The `set` will contain at most x records (20 by default, I think) in this case.
Also.. don't do it like so. Make another query to the API with isconverted filter and use the filter's count parameter.